contracts/bonding/BondingManager.sol

```diff
@@ -408,10 +415,20 @@ contract BondingManager is IBondingManager, Manager {
             transcoderTotalStake: _transcoderTotalStake
         }));
 
-        // Calculate transcoder fees including share of delegator fees based upon the amount it bonded to self
-        uint256 transcoderFeeShare = _fees.sub(delegatorsFeeShare).add(delegatorsFeeShare.mul(t.bondedAmount).div(transcoderTotalStake(_transcoder)));
-        // Update transcoder stake with fee share
-        t.bondedAmount = t.bondedAmount.add(transcoderFeeShare);
+        // Calculate transcoder fees
+        uint256 transcoderFeeShare = _fees.sub(delegatorsFeeShare);
+
+        delegators[msg.sender].bondedAmount = delegators[msg.sender].bondedAmount.add(transcoderFeeShare);
+
+        if (delegatorStatus(msg.sender) == DelegatorStatus.Unbonded) {
+            // Create new delegator if transcoder is not already a bonded delegator
+            delegators[msg.sender].delegationTarget = msg.sender;
+            delegators[msg.sender].startRound = currentRound;
+            delegators[msg.sender].delegateBlock = block.number;
+            delegators[msg.sender].withdrawRound = 0;
+            delegators[msg.sender].lastStakeUpdateRound = currentRound;
```

Comment:
I think all this logic might be repeated in `reward()` as well. Maybe a convenience method for having a new delegator be initialized in one place would be good to isolate the logic.